pkg/openstack/loadbalancer.go

```diff
@@ -712,48 +712,40 @@ func getSubnetIDForLB(network *gophercloud.ServiceClient, node corev1.Node, pref
 }
 
 // applyNodeSecurityGroupIDForLB associates the security group with all the ports on the nodes.
-func applyNodeSecurityGroupIDForLB(compute *gophercloud.ServiceClient, network *gophercloud.ServiceClient, nodes []*corev1.Node, sg string) error {
+func applyNodeSecurityGroupIDForLB(network *gophercloud.ServiceClient, nodes []*corev1.Node, sg string) error {
 	for _, node := range nodes {
-		nodeName := types.NodeName(node.Name)
-		srv, err := getServerByName(compute, nodeName)
-		if err != nil {
-			return err
-		}
-
-		listOpts := neutronports.ListOpts{DeviceID: srv.ID}
+		listOpts := neutronports.ListOpts{DeviceID: node.Spec.ProviderID}
```

Comment:
shouldn't we take into account that the ProviderID can have the following format `openstack:///a0e45d62-af9d-4453-a569-fe7094204ec6`  and extract the UUID from that?

Reply:
+1 to the question

I think it's `openstack:///` for node.Spec.ProviderID

Reply:
Whoops, yes, I fixed it locally but uploaded the wrong version. Thanks for a careful review!

Reply:
maybe we need think about how to add those securty group test into CI so we can catch 
but that's a different topic :)

Reply:
Being able to mock Gophercloud in the tests would be really cool. I'm not sure if we should wait for some Gophercloud utility or just apply CAPO's approach.

Reply:
Aside: I would also very much like to be able to unit test code which makes OpenStack calls in CPO. The CAPO approach is one possibility, but I've also started looking into ways we could do this in Gophercloud, up to and including a full simulator. This would not be a small amount of work, though.